### PR TITLE
Wallet and backup payment implementation

### PR DESCRIPTION
**Pull request by jorge**
POST v2/accounts/:account_code/billing_infos
```ruby
account = Recurly::Account.find('account-code')
billing_info = Recurly::BillingInfo.new(
  first_name: 'Lentil',
  last_name: 'Peanut',
  address1: '789 Seventy One Fake Street',
  city: 'Fake Town',
  state: 'ME',
  zip: '04011',
  country: 'US',
  number: '4111-1111-1111-1111',
  month: 11,
  year: 2049,
)
account.create_billing_info(billing_info)
```

GET v2/accounts/{account_code}/billing_infos
```ruby
account.get_billing_infos
```

GET v2/accounts/{account_code}/billing_infos/{uuid}
```ruby
account.get_billing_info('billing-info-uuid')
```

PUT v2/accounts/{account_code}/billing_infos/{billing_info_id}
```ruby
binfo = account.get_billing_info('billing-info-uuid')
binfo.update_attributes(
  city: 'Big Fake City',
  primary_payment_method: true
)
```

DELETE v2/accounts/{account_code}/billing_infos/{billing_info_id}
```ruby
binfo = account.get_billing_info('billing-info-uuid')
binfo.destroy
```

POST v2/accounts and PUT v2/accounts/{account_code}
These endpoints are documented https://developers.recurly.com/api-v2/v2.29/index.html#operation/updateAccountsBillingInfoCreditCard and https://developers.recurly.com/api-v2/v2.29/index.html#operation/updateAccount, for example.

POST v2/purchases
```ruby
purchase = Recurly::Purchase.new({
  currency: 'USD',
  collection_method: :automatic,
  billing_info_uuid: 'billing-info-uuid',
  account: {
    account_code: account.account_code,
  },
  subscriptions: [
    plan_code: plan.plan_code,
  ],
  shipping_address: valid_shipping_info
})
collection = Recurly::Purchase.invoice!(purchase)
```

POST v2/subscriptions
```ruby
subscription = Recurly::Subscription.create!(
  plan_code:            plan.plan_code,
  currency:             'USD',
  unit_amount_in_cents: 1200,
  billing_info_uuid: 'billing-info-uuid',
  account:              {
    account_code: account.account_code
  }
)
```

PUT v2/subscriptions/{uuid}
```ruby
subscription = Recurly::Subscription.find('subscription-uuid)
subscription.update_attributes(
  billing_info_uuid: 'billing-info-uuid'
)
```

PUT v2/invoices/{invoice_number}/collect
```ruby
invoice = Recurly::Invoice.find('2505)
invoice.billing_info_uuid = 'billing-info-uuid'
puts invoice.force_collect
```

### CODE REVIEW

File: lib/recurly/purchase.rb [hunk 1]
```diff
@@ -189,7 +193,6 @@ def pending!(purchase)
 
       # Allows the merchant to cancel an authorization.
       #
-      # @param transaction_uuid [String] The uuid for the transaction representing the authorization. Can typically be found at invoice_collection.charge_invoice.transactions.first.uuid.
```
Comment (joannasese) [inline lib/recurly/purchase.rb]: I think this was removed by accident.

File: lib/recurly/account.rb [hunk 2]
```diff
@@ -124,6 +124,25 @@ def build_invoice
       raise Invalid, e.message
     end
 
+    def create_billing_info(billing_info)
```
Comment (joannasese) [inline lib/recurly/account.rb:127]: I should probably include some inline documentation

File: lib/recurly/purchase.rb (hunk 1 above)
Comment (joannasese) [reply lib/recurly/purchase.rb]: @jorge and/or @naspthorp mind putting this line back in?

Comment (naspthorp) [reply lib/recurly/purchase.rb]: Whoops!  Fixed.

**Review by dtriggs-recurly — APPROVED** [review]
It looks okay to my mostly untrained eye.

**Pull request merged**